docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+    3. A `String` object will be returned and saved into the `userCommandInput` variable in `Kitchen Helper`.
+    4. The variable `userCommandInput` is being parsed into the `Parser` class as an argument for this method `Parser#parseUserCommand`.
+2. The command inserted by the user is being parsed into the `Parser` and a new `Command` object is being created. 
+    2. The variable `userCommandInput` will be identified as `deleteingredient` in the `Parser#parseUserCommand()`.The `Parser#prepareDeleteIngredient()` is being called to prepare the `userCommandInput` string to create a `DeleteIngredientCommand` object.
+    3. The `DeleteIngredientCommand` object is created with the ingredientIndex and quantity set to 4. 
+3. The Command is now being executed. 
+    2. The `DeleteIngredientCommand#execute()` will be called.
+    3. As this is a deletion by ingredient index, the `ingredientIndex` variable is not null. As the `ingredientIndex` is not null, `DeleteIngredientCommand#deleteIngredientByIndex()`.
+    4. Next, `DeleteIngredientCommand#deleteIngredient()` is called to reduce the quantity of this ingredient since the `quantity` variable is not null. 
+    6. Next, `DeleteIngredientCommand#updateNewQuantity()` will be called to update the quantity of this ingredient in our ingredients’ inventory.
+    7. Lastly, a String called `feedbackToUser`will be returned to the user to inform the user of the outcome of the command. 
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+
+##### Design Considerations
+Aspect 1: How to differentiate `deleteingredientByQuantity` and `deleteIngredient` <br>
+<br>
+Alternative 1: The `quantity` of ingredient in `DeleteIngredientCommand` constructor is set to the `quantity` that was inputted by the user. In the case where the user would like to delete an ingredient, the `quantity` variable will be set to `null`. (Current Choice)
+
+|     |     |
+|-----|-----|
+|**Pros**|Only a `quantity` variable needs to be set. This increases more convenience and no overload of constructors.|
+|**Cons**|Dependent on the variable to check if the ingredient is to be deleted. | 
+
+Alternative 2: Create 2 more constructors just for deduction of quantity for ingredients. <br>
+
+|     |     |
+|-----|-----|
+|**Pros**|This gives us more flexibility on what object can be created with different variables since there are two methods of delete of ingredients.|
+|**Cons**|There is an overload of constructors.|
+<br>
+
+#### 4.1.4. Search for ingredients based on keyword(s)
+
+The search for ingredients feature allows the user to find ingredients using a keyword in the ingredient’s list.  
+For example, `searchingredient beef` will find all the ingredients that contain `beef`.  
+
+##### Implementation  
+
+![SearchIngredientCommand](images/searchIngredientCommand1.png) 
```

Comment:
Maybe you can use an "X" to indicate the point at which the objects ceases to be used? Because at any time only 1 command is "alive/being used".. 😄